VideoAPI/Video.API/Controllers/VideoEventsController.cs

```diff
@@ -61,20 +58,57 @@ public async Task<IActionResult> PostEventAsync(ConferenceEventRequest request)
                 return NoContent();
             }
 
-            var callbackEvent = new CallbackEvent
+            var callbackEvent = MapEventRequestToEventHandlerDto(conferenceId, participantId, request);
+            await _eventHandlerFactory.Get(request.EventType).HandleAsync(callbackEvent);
+            return NoContent();
+        }
+
+        private SaveEventCommand MapEventRequestToEventCommand(Guid conferenceId, ConferenceEventRequest request)
+        {
+            GetRoomTypeEnums(request, out var transferTo, out var transferFrom);
+
+            var command = new SaveEventCommand(conferenceId, request.EventId, request.EventType,
+                request.TimeStampUtc, transferFrom, transferTo, request.Reason, request.Phone);
+            if (Guid.TryParse(request.ParticipantId, out var participantId))
+            {
+                command.ParticipantId = participantId;
+            }
+
+            command.TransferredFromRoomLabel = request.TransferFrom;
+            command.TransferredToRoomLabel = request.TransferTo;
+
+            return command;
+        }
+
+        private CallbackEvent MapEventRequestToEventHandlerDto(Guid conferenceId, Guid participantId,
```

Comment:
do we need to do a null / empty check on conferenceId and participantId?
in c# 9 new feature to take care of like this 👍 someday soon
`Method(T name!)`

Reply:
I was very inclined to update the project language but we would need to upgrade to dotnet core 5 I believe

Reply:
yup :) do we need to have null check?